Examples/Scripts/MaterialMapping/GeometryVisualisationAndMaterialHandling.py

```diff
@@ -0,0 +1,784 @@
+
+    volume_shift = [0] * len(index_to_names)
+
+    for entry in data['Surfaces']['entries']:
+        if "layer" in entry:
+            extends = []
+            vol = entry['volume']
+
+            if "sensitive" in entry:
+                continue
+
+            z_shift = 0.
+            if entry['value']['transform']['translation'] != None:
+                z_shift = entry['value']['transform']['translation'][2]
+
+            if entry['value']['type'] == "CylinderSurface":
+                extends = [0., entry['value']['bounds']['values'][0],
+                           z_shift-entry['value']['bounds']['values'][1], z_shift+entry['value']['bounds']['values'][1]]
+            else:
+                extends = [entry['value']['bounds']['values'][0], entry['value']['bounds']['values'][1],z_shift]
+
+            if "approach" in entry:
+                extends.append(entry["approach"])
+            else:
+                extends.append(-1)
+            extends.append(entry["layer"])
+
+            if entry['value']['type'] == "CylinderSurface":
+                if "approach" in entry:
+                    index_to_extends_layers_approach_cylinders[vol-1].append(extends)
+                else:
+                    index_to_extends_layers_cylinders[vol-1].append(extends)
+            else:
+                if "approach" in entry:
+                    index_to_extends_layers_approach_discs[vol-1].append(extends)
+                else:
+                    index_to_extends_layers_discs[vol-1].append(extends)
+
+        if "boundary" in entry:
+            extends = []
+            vol = entry['volume']
+
+            z_shift = 0.
+            if entry['value']['transform']['translation'] != None:
+                z_shift = entry['value']['transform']['translation'][2]
+
+            if entry['value']['type'] == "CylinderSurface":
+                extends = [0., entry['value']['bounds']['values'][0],
+                           z_shift-entry['value']['bounds']['values'][1], z_shift+entry['value']['bounds']['values'][1],entry['boundary']]
+                index_to_extends_layers_bounds_cylinders[vol-1].append(extends)
+            else:
+                extends = [entry['value']['bounds']['values'][0], entry['value']['bounds']['values'][1],z_shift,entry['boundary']]
+                index_to_extends_layers_bounds_discs[vol-1].append(extends)
+
+    # Steering the information and collect it into an output file if needed
+
+    interesting_volumes = []
+    v_index = 0
+    for elements in index_to_extends_layers_cylinders:
+        for coords in elements:
+            if v_index not in interesting_volumes:
+                interesting_volumes.append(v_index)
+            if index_to_names[v_index] not in steering_map:
+                steering_map[index_to_names[v_index]] = index_info(v_index+1)
+            steering_map[index_to_names[v_index]].layers_no_approach.append(coords[5])
+        v_index = v_index+1
```

Comment:
It is realy a question. Does it mean that the volume ID are "generated" here and reflect exactly what is done doing geometry building in c++? My concern is if the could be desynchronised.

Reply:
The module reads the json file dumped from the tracking geometry. So, yes, I would say that this is reflecting what you generate with the c++ geometry building code.
I tested this within athena for the ITk and with the standalone ITk chain and all seems consistent.

Reply:
But is this good enough? I think recently there is a functionality added to provide customised "labeler of volumes & layers" https://github.com/acts-project/acts/pull/1779. I understand the plan is to apply it also for the ITk. 

Anyway, what would we risk if the c++ and configs obtained though this script are out of sync? 
My thinking is following. If the issue due to inconsistency is evident and obvious. Fine, it will be spot, investigate and fixed quickly. 
If the issue may be subtle and visible only when doing careful validation maybe we should strive for perfect consistency?

Or maybe we should say in the doc, (take these volume & layer IDs with a grain of salt and please run GeoBuilding in DEBUG mode to re-verify them?

Reply:
The indices dumped in the JSON used as input to this module are read from the geometry, whatever is the geometry ID hook mechanism. This means that they are the exact same ones as set in the geometry building.
There is no labelling of volumes and layers in this script.
I am pretty sure this script works even after https://github.com/acts-project/acts/pull/1779 is merged, but I can of course double check it.

Reply:
oh. ok. I have missed that. Then this is non-issue. Sorry for the noise.